lib/mzgl/gl/Graphics.cpp

```diff
@@ -858,3 +541,8 @@ void Graphics::clear(float c) {
 void Graphics::clear(float r, float g, float b, float a) {
 	clear({r, g, b, a});
 }
+
+#include "OpenGLAPI.h"
+Graphics::Graphics() {
+	api = std::make_unique<OpenGLAPI>(*this);
```

Comment:
Ah okay, I see the need for the friend class now. I think a factory would be a better option - That way Graphics stays clean and the implementations are clear to behave in whatever way they need to...

Reply:
I only needed the friend thing for one private variable I wanted to share - it was the defaultFBO, I've fixed it with a dynamic_cast (may also be a bit hacky)